src/modules/navigator/mission_base.cpp

```diff
@@ -293,17 +294,22 @@ MissionBase::on_active()
 		_align_heading_necessary = false;
 	}
 
-	// replay gimbal and camera commands immediately after resuming mission
-	if (haveCachedGimbalOrCameraItems()) {
-		replayCachedGimbalCameraItems();
-	}
+	// Replay cached mission commands once the last mission waypoint is re-reached after the mission interruption.
+	// Each replay function also clears the cached items afterwards
+	if (_mission.current_seq > _replay_index)
+	{
+		// replay gimbal and camera commands immediately after resuming mission
+		if (haveCachedGimbalOrCameraItems()) {
```

Comment:
I'm not opinionated for one or the other, but when you do the `replayCachedGimbalCameraItems()` you may have the camera not yet correctly configured when the triggering starts. What's the issue with configuring it already when the Mission is resumed, and not only when the vehicle is at the correct position?

Reply:
That's a good point, I have split the replay/checks of gimbal commands from the replay/checks if the camera mode. In this way we can replay the camera commands immediately and the gimbal commands later